### ISSUE TITLE
JdbiSqliteExtension does not work with SQLite memory databases

### ISSUE THREAD

**Issue by ptomli** [jdbi/jdbi#2688]
`JdbiSqliteExtension` uses a JDBC URL of `jdbc:sqlite::memory:`.

From the SQLite memory database documentation, https://www.sqlite.org/inmemorydb.html

> The database ceases to exist as soon as the database connection is closed. Every :memory: database is distinct from every other. So, opening two database connections each with the filename ":memory:" will create two independent in-memory databases.

This means that every connection via `JdbiSqliteExtension` gets a new database, which will not include any data from previous queries, including `JdbiExtensionInitializer` efforts.

Multiple connections can use the same memory database if they use shared memory, by specifying `cache=shared` on the JDBC URL. Further, multiple shared memory databases can be used if the URL names a file and indicates `mode=memory`, eg:

`jdbc:sqlite:file:memory?mode=memory&cache=shared`

Please see the "In-memory Databases And Shared Cache" section of the linked documentation.

I do not believe that `JdbiSqliteExtension` can work as intended in its current form.

I have been able to work around this locally by extending the extension.

```
public class MyJdbiSqliteExtension extends JdbiSqliteExtension {
    @Override
    public String getUrl() {
       return "jdbc:sqlite:file:example?mode=memory&cache=shared";
    }
}

@RegisterExtension
public static JdbiExtension extension = new MyJdbiSqliteExtension()
            .withPlugin(new SQLitePlugin())
            .withInitializer((ds, handle) -> { ... });
```

Probably unrelated, but for some reason I can't determine, I'm not able to use `@RegisterExtension` and `JdbiExtension` on an instance variable, only as a class/static variable.

**Comment by stevenschlansker** [comment]
Thanks for reporting this.

**Comment by stevenschlansker** [comment]
Thanks @ptomli , if you have a sec please review #2707 as a candidate fix.

**Comment by ptomli** [comment]
I'm not in a position to check that out to test, but it does on the face of it look like it should work.

**Comment by stevenschlansker** [comment]
fixed in 3.46.0
